src/server/getters/get-markets-info.ts

```diff
@@ -1,37 +1,31 @@
 import { each } from "async";
 import BigNumber from "bignumber.js";
 import * as Knex from "knex";
-import { Address, MarketsRow, OutcomesRow, UIMarketInfo, UIOutcomeInfo, ErrorCallback } from "../../types";
+import { Address, MarketsRow, OutcomesRow, UIMarketInfo, UIMarketsInfo, UIOutcomeInfo, ErrorCallback } from "../../types";
 import { reshapeOutcomesRowToUIOutcomeInfo, reshapeMarketsRowToUIMarketInfo } from "./get-market-info";
 
-export interface UIMarketsInfo {
-  [marketID: string]: UIMarketInfo;
-}
-
 export function getMarketsInfo(db: Knex, universe: Address|null|undefined, marketIDs: Array<Address>|null|undefined, callback: (err?: Error|null, result?: UIMarketsInfo) => void): void {
-  let query: string = "SELECT * FROM markets WHERE";
-  let queryParams: Array<Address|Array<Address>>;
+  let marketsQuery: Knex.QueryBuilder = db.select().from("markets").orderBy("creationTime");
+
   if (universe == null && marketIDs == null) {
     return callback(new Error("must include universe or marketIDs parameters"));
-  } else if (universe != null && marketIDs != null) {
-    query = `${query} universe = ? AND "marketID" IN (??)`;
-    queryParams = [universe!, marketIDs!];
-  } else if (universe != null) {
-    query = `${query} universe = ?`;
-    queryParams = [universe!];
-  } else {
-    query = `${query} "marketID" IN (??)`;
-    queryParams = [marketIDs!];
   }
-  db.raw(query, queryParams).asCallback((err?: Error|null, marketsRows?: Array<MarketsRow>): void => {
+  if (universe != null) {
+    marketsQuery = marketsQuery.where( {universe } );
+  }
+  if (marketIDs != null) {
+    marketsQuery = marketsQuery.whereIn( "marketID", marketIDs );
+  }
+
+  marketsQuery.asCallback((err?: Error|null, marketsRows?: Array<MarketsRow>): void => {
     if (err) return callback(err);
     if (!marketsRows || !marketsRows.length) return callback(null);
-    const marketsInfo: UIMarketsInfo = {};
+    const marketsInfo: UIMarketsInfo = [];
     each(marketsRows, (marketsRow: MarketsRow, nextMarketsRow: ErrorCallback): void => {
       db.raw(`SELECT * FROM outcomes WHERE "marketID" = ?`, [marketsRow.marketID]).asCallback((err?: Error|null, outcomesRows?: Array<OutcomesRow>): void => {
```

Comment:
Need to change this to the QB form as well or it won't return the right results against PGSql

Reply:
Actually, take a look at writing this query as a join btw `markets` and `outcomes`.

Reply:
Ha, can't believe i missed that raw call there, good catch.

Reply:
it's a one-to-many table, it seems cleaner to use separate queries, or else we're returning the market over and over, collapsing it in the code, or using an aggregate sql function. Unless you know a better way with Knex?